skiko/src/jvmMain/kotlin/org/jetbrains/skiko/SkiaLayer.jvm.kt

```diff
@@ -150,59 +150,58 @@ actual open class SkiaLayer internal constructor(
 
     fun attachTo(jComponent: JComponent) {
         jComponent.add(this)
-        backedLayer.addMouseListener(object : MouseAdapter() {
-            override fun mousePressed(e: MouseEvent?) {
-                e!!
-                skikoView?.onPointerEvent(
-                    SkikoPointerEvent(e.x.toDouble(), e.y.toDouble(),
-                        if (e.button == 1) SkikoMouseButtons.LEFT else SkikoMouseButtons.RIGHT,
-                        SkikoPointerEventKind.DOWN,
-                        e
-                    )
-                )
+    }
+
+    fun addView(view: SkikoView) {
+        skikoView = view
+        addMouseListener(object : MouseAdapter() {
+            override fun mousePressed(e: MouseEvent) {
+                skikoView?.onPointerEvent(toSkikoEvent(e))
+            }
+            override fun mouseReleased(e: MouseEvent) {
+                skikoView?.onPointerEvent(toSkikoEvent(e))
+            }
+            override fun mouseEntered(e: MouseEvent) {
+                skikoView?.onPointerEvent(toSkikoEvent(e))
             }
-            override fun mouseReleased(e: MouseEvent?) {
-                e!!
-                skikoView?.onPointerEvent(
-                    SkikoPointerEvent(e.x.toDouble(), e.y.toDouble(),
-                        if (e.button == 1) SkikoMouseButtons.LEFT else SkikoMouseButtons.RIGHT,
-                        SkikoPointerEventKind.UP,
-                        e
-                    )
-                )
+            override fun mouseExited(e: MouseEvent) {
+                skikoView?.onPointerEvent(toSkikoEvent(e))
             }
         })
-        backedLayer.addMouseMotionListener(object : MouseMotionAdapter() {
-            override fun mouseMoved(e: MouseEvent?) {
-                e!!
-                skikoView?.onPointerEvent(
-                    SkikoPointerEvent(e.x.toDouble(), e.y.toDouble(),
-                        SkikoMouseButtons.NONE,
-                        SkikoPointerEventKind.MOVE,
-                        e
-                    )
-                )
+
+        addMouseMotionListener(object : MouseMotionAdapter() {
```

Comment:
One of things we shall be doing is let view to dynamically control which events it is subscribed to, especially high frequency ones, like move.

Reply:
I don't think it is an issue. Just subscribing and converting an event to another type shouldn't be a bottleneck in any application.

Reply:
Think pointer move events - they're very high frequency, and if we'll always subscribe to those it will mean a lot of unneeded code executed. There are other events which are pretty high frequency, i.e. motion sensor events, which we only shall get if explicitly requested.

Reply:
> Think pointer move events - they're very high frequency, and if we'll always subscribe to those it will mean a lot of unneeded code executed

It looks scary, but what in reality? Will just conversions cause any performance issues. event it is just unneeded code? 

This kind of optimization will severely complicate the code in skiko and in Compose. It is probably even impossible to do it in Compose, with the current commonMain/skikoMain code without huge API redesign.

Reply:
It's not the conversion itself, but more like registering for particular event would lead to more frequent app resume and thus huge battery drain on device. And APIs are not hard to make, could prototype it separately, if it's unclear what I mean.